mmv1/products/compute/api.yaml

```diff
@@ -11981,6 +11981,149 @@ objects:
               If set to true, any accompanying query portion of the original URL is removed prior
               to redirecting the request. If set to false, the query portion of the original URL is
               retained.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'defaultRouteAction'
+        description: |
+          defaultRouteAction takes effect when none of the hostRules match. The load balancer performs advanced routing actions, such as URL rewrites and header transformations, before forwarding the request to the selected backend. If defaultRouteAction specifies any weightedBackendServices, defaultService must not be set. Conversely if defaultService is set, defaultRouteAction cannot contain any weightedBackendServices.
+          Only one of defaultRouteAction or defaultUrlRedirect must be set.
+          URL maps for Classic external HTTP(S) load balancers only support the urlRewrite action within defaultRouteAction.
+          defaultRouteAction has no effect when the URL map is bound to a target gRPC proxy that has the validateForProxyless field set to true.
+        properties:
+          - !ruby/object:Api::Type::Array
+            name: 'weightedBackendServices'
```

Comment:
Should this field be required? Is there a situation where a user would define defaultRouteAction but *not* weightedBackendServices?

Reply:
It's a bit confusing, but I believe there is a situation like that. I think someone could set [`(regionUrlMap).defaultService`](https://cloud.google.com/compute/docs/reference/rest/v1/regionUrlMaps#UrlMap.FIELDS.inlinedField_5) to specify a single service that requests are sent to (versus the weighted list of options) and can then set fields inside `defaultRouteAction` like `(regionUrlMap).defaultRouteAction.retryPolicy` to define how the request will be handled. In that scenario, the user wouldn't be able to set `weightedBackendServices` because `defaultService` is specified.

I'll have a go testing this with the API to check 🏃‍♀️

Reply:
Yeah that seems to be the case, I get this error when specifying both fields:


> Error: Error creating RegionUrlMap: googleapi: Error 400: Invalid value for field 'resource.defaultRouteAction.weightedBackendServices[0]': '{  "backendService": "projects/[projectID]/regions/us-central1/backendServices/homeq1w6iebz...'. Exactly one of default backend service or default route action weighted backend service must be specified for the url map, invalid

In the final provider code there's schema properties like this enforcing the different mutually exclusive fields in the API:

```
ExactlyOneOf: []string{"default_service", "default_url_redirect", "default_route_action.0.weighted_backend_services"},
```